### DISCUSSION TITLE
Support for HuggingFace TGI

### DISCUSSION THREAD

**Discussion by kengz** [open-webui/open-webui#2206] [General]
**Is your feature request related to a problem? Please describe.**
Existing [OpenAI API Endpoints](https://docs.openwebui.com/tutorial/openai) config doesn't work for [HuggingFace TGI](https://github.com/huggingface/text-generation-inference). TGI serves one model at a time, but has no `/models` endpoint (see [TGI API doc here](https://huggingface.github.io/text-generation-inference/)) that WebUI requires - also it doesn't care what "model" is passed in request param. As a result, configuring with TGI fails.

**Describe the solution you'd like**
Current config uses `OPENAI_API_BASE_URLS, OPENAI_API_KEYS`, but for TGI, it needs to explicitly specify which URL maps to which model, e.g. `Llama3 -> http://llama3.tgi.my-example.com/v1`, `Gemma -> http://gemma.tgi.my-example.com/v1`, etc.

**Comment by cheahjs** [comment]
Open WebUI includes [LiteLLM](https://docs.litellm.ai/docs/providers/huggingface) as a proxy between various LLM providers and an OpenAI-compatible endpoint. You can add a TGI endpoint to LiteLLM via the UI.

![image](https://github.com/open-webui/open-webui/assets/818368/fbb1566f-bbb8-4be7-b335-370dd24fde01)
